packages/lesswrong/server/fmCrosspost/resolvers.ts

```diff
@@ -36,15 +44,25 @@ export const makeCrossSiteRequest = async <RouteName extends ValidatedPostRouteN
         "User-Agent": crosspostUserAgent,
       },
       body: JSON.stringify(body),
+      signal: controller.signal,
     });
   } catch (e) {
-    if (e.cause.code === 'ECONNREFUSED' && e.cause.port === 4000) {
+    clearTimeout(timeoutId); // Clear the timeout when an error occurs
+
+    if (e.name === 'AbortError') {
+      throw new Error('Request timed out');
+    }
```

Comment:
Can we make this a nicer error condition? It's kinda an expected one. We can decide to not show this recent discussion card.

Reply:
I've made it an ApiError at least. I tried making it hide the recent discussion card but for some reason the error doesn't come through. It's probably easy to fix but I'm giving up in order to get this merged before this (probably) happens again this evening